src/include/sof/common.h

```diff
@@ -14,20 +14,25 @@
 /* Align the number to the nearest alignment value */
 #define IS_ALIGNED(size, alignment) ((size) % (alignment) == 0)
 
-#define is_non0_power_of_2(x) ((x) && !((x) & ((x) - 1)))
+/* Treat zero as a special case because it wraps around */
```

Comment:
to me this comment confuses me more than helps to understand, but maybe it's just me :-) The macro seems rather clear by itself: 0 isn't a power of 2 but just using `!(x & (x-1))` would falsely return `true` on `x=0`. Simple enough IMHO.

Reply:
Why does `!(x & (x-1))` work for everything except zero?

Reply:
The RHS wont be evaluated if LHS is 0, hence the macro result will be valid.
@marc-hb got some red in CI, so will rerun as there have been some server reboots.

Reply:
This was a rhetorical question to explain the answer I'm adding as a comment :-)